app/ante/tx_size_test.go

```diff
@@ -0,0 +1,174 @@
+package ante_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v3/app"
+	"github.com/celestiaorg/celestia-app/v3/app/ante"
+	"github.com/celestiaorg/celestia-app/v3/pkg/appconsts"
+	v3 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v3"
+	testutil "github.com/celestiaorg/celestia-app/v3/test/util"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/crypto/types/multisig"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	xauthsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+func setup() (*app.App, sdk.Context, client.Context, error) {
+	app, _, _ := testutil.NewTestAppWithGenesisSet(app.DefaultConsensusParams())
+	ctx := app.NewContext(false, tmproto.Header{})
+	app.AccountKeeper.SetParams(ctx, authtypes.DefaultParams())
+	ctx = ctx.WithBlockHeight(1)
+
+	// Set up TxConfig.
+	encodingConfig := simapp.MakeTestEncodingConfig()
+	// We're using TestMsg encoding in the test, so register it here.
+	encodingConfig.Amino.RegisterConcrete(&testdata.TestMsg{}, "testdata.TestMsg", nil)
+	testdata.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+
+	clientCtx := client.Context{}.
+		WithTxConfig(encodingConfig.TxConfig)
+
+	return app, ctx, clientCtx, nil
+}
+
+func TestConsumeGasForTxSize(t *testing.T) {
+	app, ctx, clientCtx, err := setup()
+	require.NoError(t, err)
+	var txBuilder client.TxBuilder
+
+	// keys and addresses
+	priv1, _, addr1 := testdata.KeyTestPubAddr()
+
+	// msg and signatures
+	msg := testdata.NewTestMsg(addr1)
+	feeAmount := testdata.NewTestFeeAmount()
+	gasLimit := testdata.NewTestGasLimit()
+
+	cgtsd := ante.NewConsumeGasForTxSizeDecorator(app.AccountKeeper)
+	antehandler := sdk.ChainAnteDecorators(cgtsd)
+
+	testCases := []struct {
+		name  string
+		sigV2 signing.SignatureV2
+	}{
+		{"SingleSignatureData", signing.SignatureV2{PubKey: priv1.PubKey()}},
+		{"MultiSignatureData", signing.SignatureV2{PubKey: priv1.PubKey(), Data: multisig.NewMultisig(2)}},
```

Comment:
I think it would be beneficial to have test cases for the different app versions to test this logic:

```go
	// For app v2 and below we should get txSizeCostPerByte from auth module
	if ctx.BlockHeader().Version.App <= v2.Version {
		ctx.GasMeter().ConsumeGas(params.TxSizeCostPerByte*cost, "txSize")
	} else {
		// From v3 onwards, we should get txSizeCostPerByte from appconsts
		ctx.GasMeter().ConsumeGas(appconsts.TxSizeCostPerByte(v3.Version)*cost, "txSize")
```

which seems like the major difference from upstream

Reply:
I'd say after we update the application to v3 so i can just use latest version

Reply:
1. Why not update the latest version constant to v3 in this PR? 
2. Why should the latest version constant block unit tests in this PR? Without unit tests, it's not easy to determine if this feature works as expected.

Reply:
1. I didn't want to update LatestVersion constant because we haven't properly upgraded the app to v3 and iirc @cmwaters was planning on doing it.
3. It's not really blocking, to my understanding we hadn't yet bumped the app to the latest version hence why I had a follow up ticket for looking back on the implementation + I manually verified that both cases worked. I updated the test to test v2.Version and v3.Version so we don't have to have reservations around testing. I'll go back to this and change all v3.Version references to LatestVersion

Reply:
https://github.com/celestiaorg/celestia-app/commit/2ddaa9f97df64e703e04ce12378b607963ef6265

Reply:
Added both versions now @rootulp